csrc/multidevice/utils.cpp

```diff
@@ -228,7 +228,7 @@ IterDomain* getShardedIterDomain(
     NVF_THROW("Unexpected parallel type: ", parallel_type);
   }();
 
-  for (auto&& [index, id] : enumerate(domain)) {
+  for (auto&& [index, id] : enumerate(domain | TensorDomain::kNoReductions)) {
```

Comment:
For reduce-scatter outputs, we may return the `rDIDx` in unshardedSizes if it is ordered before `iDIDx` in the loop domain. This leads to incorrect shape deduction. For e.g.: `LowerCollectiveTest.NoncontigReduceScatter`